m2cgen/interpreters/interpreter.py

```diff
@@ -203,9 +230,15 @@ def interpret_tanh_expr(self, expr, **kwargs):
         nested_result = self._do_interpret(expr.expr, **kwargs)
         return self._cg.function_invocation(self.tanh_function_name, nested_result)
 
-    def _wrap_infix_expr(self, expr, left_precedence, right_precedence):
+    def _wrap_infix_expr(self, expr, left_precedence, right_precedence, right_is_associative):
         wrap = left_precedence is not None and left_precedence > expr.precedence
-        wrap |= right_precedence is not None and right_precedence >= expr.precedence
+        wrap |= (
```

Comment:
Maybe:
```
wrap |= right_precedence is not None and right_precedence > expr.precedence
wrap |= right_precedence is not None and right_precedence = expr.precedence and ...
```
for readability?

Reply:
TBH, I intentionally hurt readability for the sake of code generation speed.
With current long complex expression we take benefits from the fact that `and` and `or` in Python are short-circuit operators:
https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not

With your advice about comparison with `False` above, I think, this expression becomes quite readable:
```python
    if right_precedence is not None:
        wrap |= (
            right_precedence > expr.precedence
            or (right_precedence == expr.precedence and right_is_associative is False)
        )
```
WDYT?